src/client/texturesource.cpp

```diff
@@ -27,41 +27,17 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "texturepaths.h"
 #include "imagegen.h"
 
-/*
-	Stores internal information about a texture.
-*/
-
+// Stores internal information about a texture.
 struct TextureInfo
 {
 	std::string name;
-	video::ITexture *texture;
-	std::set<std::string> sourceImages;
-
-	TextureInfo(
-			const std::string &name_,
-			video::ITexture *texture_=NULL
-		):
-		name(name_),
-		texture(texture_)
-	{
-	}
+	video::ITexture *texture = nullptr;
 
-	TextureInfo(
-			const std::string &name_,
-			video::ITexture *texture_,
-			std::set<std::string> &&sourceImages_
-		):
-		name(name_),
-		texture(texture_),
-		sourceImages(std::move(sourceImages_))
-	{
-	}
+	// Stores source image names which ImageGen::generateImage used.
+	std::set<std::string> sourceImages{};
```

Comment:
pointless initialization

Reply:
No, this is needed such that `m_textureinfo_cache.emplace_back(TextureInfo{""});` can use the initializer list constructor.

Reply:
I can't find a difference: https://godbolt.org/z/zWMK9eGbP

Reply:
Well, I get this warning when I compile the code without the initialization.
```
.../minetest/src/client/texturesource.cpp:195:56: Warnung: Initialisierung für Element »TextureInfo::sourceImages« fehlt [-Wmissing-field-initializers]
  195 |         m_textureinfo_cache.emplace_back(TextureInfo{""});
```

Reply:
Add `-Wextra` to your compiler flags at godbolt and you will get the same warning.

Reply:
If you don't like it to be initialized in the struct, I can remove it and initialize it elsewhere.
i.e. use `TextureInfo{"", nullptr, std::set<std::string>{}}` instead.

Reply:
Huh, weird. That would be way less readable so no of course not.